src/text/text_formatter.rs

```diff
@@ -0,0 +1,401 @@
+use crate::raw_symbol_token_ref::AsRawSymbolTokenRef;
+use crate::types::timestamp::Precision;
+use crate::value::owned::{OwnedSequence, OwnedStruct, OwnedSymbolToken};
+use crate::value::{Sequence, Struct, SymbolToken};
+use crate::{Decimal, Integer, IonResult, IonType, RawTextWriter, Timestamp};
+use chrono::{DateTime, Datelike, FixedOffset, NaiveDateTime, TimeZone, Timelike};
+use std::convert::TryInto;
+use std::io::BufWriter;
+
+pub struct IonValueFormatter<'a, W: std::io::Write> {
+    pub(crate) output: &'a mut W,
+    pub(crate) string_escape_codes: Vec<String>,
+}
+
+impl<'a, W: std::io::Write> IonValueFormatter<'a, W> {
+    pub(crate) fn format_annotations(
+        &mut self,
+        annotations: &Vec<OwnedSymbolToken>,
+    ) -> IonResult<()> {
+        for annotation in annotations {
+            self.format_symbol(annotation.text().unwrap())?;
```

Comment:
Rather than calling `text().unwrap()` (which could panic), I think we should implement `Display` for `OwnedSymbolToken` and have it write out either `foo` or `$23` as its able to.